### ISSUE TITLE
Unable to build az_iot_hub_client samples on windows

### ISSUE THREAD

**Issue by YrnehEpoc** [Azure/azure-sdk-for-c#2741]
**Describe the bug**
I am attempting to build and run the iot hub samples on windows. I'm following the instructions [here](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_windows.md#iot-hub-telemetry-sample), but I run into an issue on the final step, [step 5](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_windows.md#configure-and-run-the-samples) When I try to build and run the project by pressing F5, I get this issue: 
![Screenshot 2024-02-12 134751](https://github.com/Azure/azure-sdk-for-c/assets/62447472/ebe3301b-c103-42ff-9b5d-babcf12963a5)

 

**To Reproduce**
Follow [these instructions](https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/iot/docs/how_to_iot_hub_samples_windows.md) on a windows machine
 

**Expected behavior**
I'm unsure but I would expect the solution to build .exe files that I can then run to test the functionality of the sdk 

**Setup (please complete the following information):**

- OS: Windows 10
- IDE: Microsoft Visual Studios 2019
- azure-sdk-for-c version 1.5.0 

**Information Checklist**
Please make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report

- [ x] Bug Description Added
- [ x] Repro Steps Added
- [x ] Setup information Added

**Comment by vaavva** [comment]
@YrnehEpoc Hello! Have you selected the project as the startup project? That seems to be what usually causes this error message. https://learn.microsoft.com/en-us/answers/questions/63937/staticlib-lib-is-not-a-valid-win32-application-err

**Comment by YrnehEpoc** [comment]
Thank you @vaavva, 
Yes I've selected the project as the startup project (right click on az_iot_hub, click on select as startup project).

![image](https://github.com/Azure/azure-sdk-for-c/assets/62447472/aaeedd86-cf0e-4854-bfa9-41bd38460c70)

I've also followed the link and added the .lib file path and name as specified. Maybe I'm not understanding the azure-sdk-for-c instructions. If this is just building the static libraries, where are the executable's that need to be linked to them?
